multicluster/controllers/multicluster/member/gateway_controller.go

```diff
@@ -77,7 +78,23 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	var err error
 	var commonArea commonarea.RemoteCommonArea
 	commonArea, r.localClusterID, err = r.commonAreaGetter.GetRemoteCommonAreaAndLocalID()
-	if commonArea == nil {
+	gw := &mcv1alpha1.Gateway{}
+	var isNotFound bool
+	if err := r.Client.Get(ctx, req.NamespacedName, gw); err != nil {
```

Comment:
You said you changed to re-sync all resources after CommonArea is initialized? In that case, we can just ignore any event here if CommonArea is nil?

Reply:
Refined this part to return immediately when the ClusterSet is not ready.